File: RB120/04_practice_problems/easy_02/01.rb
# Exercise 01

class Oracle
  def predict_the_future
    "You will " + choices.sample
  end

  def choices
    ["eat a nice lunch", "take a nap soon", "stay at work late"]
  end
end

# What is the result of executing the following code:

oracle = Oracle.new
p oracle.predict_the_future

# We will se returned a string object composed by the concatenation of the string object
# `"You will"` and a random string object from the array on line 9.

File: exercises/js_fundamentals/medium_01/01.js
// Exercise 01

// What will each line of the following code return? 
// Don't run the code until after you have tried to answer.

console.log((false && undefined)); // false 
console.log((false || undefined)); // undefined
console.log(((false && undefined) || (false || undefined))); // undefined
console.log(((false || undefined) || (false && undefined))); // false
console.log(((false && undefined) && (false || undefined))); // false
console.log(((false || undefined) && (false && undefined))); // undefined
console.log(('a' || (false && undefined) || '')); // 'a'
console.log(((false && undefined) || 'a' || '')); // 'a'
console.log(('a' && (false || undefined) && '')); // undefined
console.log(((false || undefined) && 'a' && '')); // undefined